[it] Localization Shift Left (#1528)
cncf/glossary: content/it/shift-left.md

---
title: Shift Left
status: Completed
category: Concetto
tags: ["metodologia", "approccio cloud native"]
---

## Cos'è

> _Left_ in _Shift Left_ (letteralmente "sposta a sinistra") si riferisce alle fasi precedenti di un ciclo di vita dello sviluppo del software, se si immagina il ciclo di vita come una linea llungo la quale le fasi si susseguono da sinistra a destra.

Reviewer: ```suggestion
_Left_ in _Shift Left_ (letteralmente "spostamento a sinistra") si riferisce alle fasi precedenti di un ciclo di vita dello sviluppo del software, se si immagina il ciclo di vita come una linea lungo la quale le fasi si susseguono da sinistra a destra.
```

> _Shift Left_ è l'approccio per il quale l'implementazione di test, di pratiche di sicurezza o di altre accorgimenti viene anticipata nelle prime fase del ciclo di vita dello sviluppo piuttosto che integrata solo alla fine dello stesso.

Reviewer: ```suggestion
_Shift Left_ è l'approccio per il quale l'implementazione di test, di pratiche di sicurezza o di altre accorgimenti viene anticipata nelle prime fasi del ciclo di vita dello sviluppo piuttosto che integrata solo alla fine dello stesso.
```

> Per quanto fosse originariamente utilizzato per riferirsi al processo di testing anticipato rispetto al ciclo nel suo complesso, _Shift Left_ può ora essere applicato anche ad altri aspetti dello sviluppo del software e di[DevOps](it/devops/), quali la sicurezza e il deployment. 

Reviewer: ```suggestion
Per quanto fosse originariamente utilizzato per riferirsi al processo di testing anticipato rispetto al ciclo nel suo complesso, _Shift Left_ può ora essere applicato anche ad altri aspetti dello sviluppo del software e di [DevOps](/it/devops/), quali la sicurezza e il deployment. 
```

## Quali problematiche affronta

Problemi di sicurezza, bug e difetti del software possono essere più difficili e costosi da risolvere se vengono scoperti in una fase avanzata del ciclo di sviluppo o dopo la deployment, soprattutto se il software è già stato distribuito in produzione.

## In che modo aiuta

Adottando una mentalità _Shift Left_ per lo sviluppo del software, i team possono implementare testing e pratiche di sicurezza durante l'intero ciclo di vita dello sviluppo perché tale responsabilità viene condivisa dall'intero team - dagli ingegneri ai responsabili della qualità fino alle operations - cosicché tutti abbiano un ruolo proattivo nel garantire la stabilità e la sicurezza di un'applicazione.

> Inoltre, lo spostamento a sinistra facilita il miglioramento continuo, pruovendo un approccio allo sviluppo [agile](it/agile-software-development/) piuttosto che _waterfall_ (a cascata).

Reviewer: ```suggestion
Inoltre, lo spostamento a sinistra facilita il miglioramento continuo, seguendo un approccio allo sviluppo [agile](/it/agile-software-development/) piuttosto che _waterfall_ (a cascata).
```
